pysat/_meta.py

```diff
@@ -271,7 +271,7 @@ def drop(self, names):
         """Drops variables (names) from metadata."""
 
         # drop lower dimension data
-        self._data = self._data.drop(names, axis=0)
+        self.data = self._data.drop(names, axis=0)
```

Comment:
Is this correct?

Reply:
Yes. I double checked. properties on Meta, like .units_label, or .data are always modifiable. 
._data, or user attributes, can only be changed if mutable=True. Thanks for the note.

Reply:
So... it is on purpose that there's both a hidden and visible variable named `data`?  If so that is confusing, because I can't easily see from their name why we need both of them.

Reply:
The .data ._data thing originally came about since meta operated with assumption of lowercase names. The property .data was there to ensure that if a user ever assigned to it, the real storage ._data would get a version with lower case names. Its a technique that enables making variables public while still being able to ensure it meets certain conditions.

The change here is a bug fix transfer. Issue #497 is geared towards reviewing the content.

Reply:
So, 'data' holds upper and lowercase names while '_data' only holds lowercase names?  What happens if a user specified 'TimePiece' and 'timepiece' as separate things?  

Also, I looked at the file to gain a better understanding and this needs to be explained in the comments.

Reply:
the public `data` is a fiction, it just maps to `._data`. If a user does, `meta.data = something`, this would call the property method which would receive something as input and the method would be responsible for processing the input as appropriate and assigning it to `._data`.

The functionality of meta with regard to case is described in the Notes section of the class docstring.  The meta object would treat 'TimePiece' and 'timepiece' as the same thing.

Fixing established parts of meta is out of scope for this pull.